src/classes/HH_ManageHH_CTRL.cls

```diff
@@ -136,7 +136,21 @@ public with sharing class HH_ManageHH_CTRL {
     * @return PageReference the original calling page's URL.
     */
     public PageReference save() {
-        update hh;
+        try {
+            // we've already saved the data for the contacts in the household.
+            // now we only need to save the account fields an update, and we don't want
+            // any of the normal account trigger work to run again.  specifically,
+            // we may have updated the names and greetings, because the contacts have changed,
+            // but the user still may want auto-naming turned on.  our normal trigger flow
+            // would think the user updated the name from what it was, and thus it should be
+            // excluded from auto-naming!
+            HH_ProcessControl.inFutureContext = true;
```

Comment:
This all seems like the wrong way to fix this, even if it is the least intrusive.  What do you think?  Reread your code comment here; does this seem like the way the system should be working?

Is this a temporary fix, or just the way it is supposed to work from now on?